configuration.rst

```diff
@@ -599,6 +599,15 @@ The values of these options are resolved at runtime (only once per request, to
 not impact performance) so you can change the application behavior without having
 to clear the cache.
 
+.. caution::
+
+    Environment variables are designed to store information that can
+    dynamically change in a production environment, such as an API
+    key that may have an expiration date. This information could be updated
+    without having to redeploy the application nor clear the cache. Thus, not
+    everything is a good candidate for env vars, and
+    :ref:`parameters <configuration-parameters>` should be used in other cases.
```

Comment:
```suggestion
    :ref:`parameters <configuration-parameters>` should be challenged with environment configs.
```

Reply:
I don't know, I'm always afraid with the `this should be challenged` wording. I always have the feeling it is very vague. Does anyone else have thoughts on this?

Reply:
oki got your point now